core/src/action.rs

```diff
@@ -87,6 +87,27 @@ impl Hash for ActionWrapper {
     }
 }
 
+///This describes a key for the actions
+#[derive(Clone, PartialEq, Debug, Serialize, Eq, Hash)]
+pub enum QueryKey {
+    Entry(GetEntryKey),
+    Links(GetLinksKey),
+}
+
+///This is a payload for the Get Method
+#[derive(Clone, PartialEq, Debug, Serialize)]
+pub enum QueryPayload {
+    Entry,
+    Links((Option<CrudStatus>, GetLinksNetworkQuery)),
+}
+
+///This is a payload from a response
+#[derive(Clone, PartialEq, Debug, Serialize)]
+pub enum RespondQueryPayload {
+    Entry(Option<EntryWithMetaAndHeader>),
+    Links((GetLinksNetworkResult, String, String)),
+}
```

Comment:
I don't think we need these two `Payload*` enum anymore, or do we? If we would use `NetworkQuery` and `NetworkQueryResult` (in `core/src/network/query.rs`) we also wouldn't need to translate these types over into `NetworkQuery*` in the according reducer below.